unleash/Services/ToggleService.swift

```diff
@@ -0,0 +1,45 @@
+//
+//  ToggleService.swift
+//  Unleash
+//
+//  Copyright © 2019 Silvercar. All rights reserved.
+//
+
+import Foundation
+import PromiseKit
+
+struct ToggleService: ToggleServiceProtocol {
+    private let appName: String
+    private let instanceId: String
+    
+    init(appName: String, instanceId: String) {
+        self.appName = appName
+        self.instanceId = instanceId
+    }
+}
+
+protocol ToggleServiceProtocol {
+    func fetchToggles(url: URL) -> Promise<Features>
+}
+
+extension ToggleService {
+    func fetchToggles(url: URL) -> Promise<Features> {
+        let togglesUrl = url.appendingPathComponent("/client/features")
+        return firstly {
+            URLSession.shared.dataTask(.promise, with: try makeUrlRequest(url: togglesUrl)).validate()
+            }.map {
+                return try JSONDecoder().decode(Features.self, from: $0.data)
+        }
+    }
+    
+    private func makeUrlRequest(url: URL) throws -> URLRequest {
+        var request = URLRequest(url: url)
+        request.httpMethod = "GET"
+        request.addValue("application/json", forHTTPHeaderField: "Content-Type")
+        request.addValue("application/json", forHTTPHeaderField: "Accept")
+        request.addValue("UNLEASH-APPNAME", forHTTPHeaderField: self.appName)
```

Comment:
Are these 3 headers backwards? Should the value be first, then the header?

Reply:
Oops, I'll fix that.